### PR TITLE
[SERF-1593] Add PubSub configuration

### PR DESCRIPTION
**Pull request by Neurostep**
## Description

In this PR we introduce basic configuration for `PubSub` application services. For now, the configuration contains only Kafka-related settings.

## Checklist

- [x] Prefixed the PR title with the JIRA ticket code 
- [x] Performed simple, atomic commits with [good commit messages][commit messages]
- [x] Verified that the commit history is linear and commits are squashed as necessary
- [ ] Thoroughly tested the changes in `development` and/or `staging`
- [ ] Updated the `README.md` as necessary

## Related links

* [SERF-1593](https://scribdjira.atlassian.net/browse/SERF-1593)
* https://github.com/scribd/go-chassis/pull/47

[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Review by fotos — DISMISSED** [review]
:+1: LGTM 🚀 

Left a few nit remarks – they can be safely ignored. 😄

File: pkg/pubsub/testdata/config/pubsub.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+common: &common
+  kafka:
+
+test: &test
+  <<: *common
+  kafka:
+    # Set by APP_SETTINGS_KAFKA_BROKER_URLS env variable
+    broker_urls:
+      - "localhost:9092"
```
[the remark this answers is not in the document]
Comment (Neurostep) [reply pkg/pubsub/testdata/config/pubsub.yml:9]: @fotos it works with `, ` comma separation. For example, in [web-analytics-ingestion](https://github.com/scribd/terraform-core-platform-chassis-services/blob/1e6f65a4e9dfb154ea19c9f655d413d81655e57d/applications.tf#L46) it works without problems

File: pkg/pubsub/testdata/config/pubsub.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    # Set by APP_SETTINGS_KAFKA_CERT_PEM_KEY env variable
+    cert_pem_key: "pem key"
+    # Set by APP_SETTINGS_KAFKA_MAX_ATTEMPTS env variable
+    max_attempts: 3
+    write_timeout: "10s"
+    topic: "test-topic"
+    security_protocol: "ssl"
+
+development:
+  <<: *test
```
[the remark this answers is not in the document]
Comment (Neurostep) [reply pkg/pubsub/testdata/config/pubsub.yml:28]: This is done for testing purposes only, in `go-chassis` we don't do this of course. Having identical `tests` and `development` here is good for local development (running tests locally without needs to provide `APP_ENV` variable)

File: pkg/pubsub/testdata/config/pubsub.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fotos) [reply pkg/pubsub/testdata/config/pubsub.yml:9]: Aha… got it. It's our convention, not something supported by Viper. :+1:

File: pkg/pubsub/testdata/config/pubsub.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (fotos) [reply pkg/pubsub/testdata/config/pubsub.yml:28]: When running `mage test` the `APP_ENV` is [already set to `test`](https://github.com/scribd/go-sdk/blob/9bd7855571f32037f659efc10ae845c301ff8723/magefile.go#L17-L24). How are you running the tests? 🤔

[the remark this answers is not in the document]
Comment (Neurostep) [reply pkg/pubsub/testdata/config/pubsub.yml:28]: @fotos When developing I often use IDE capability to run test cases right from the IDE UI/UX, it is not always convenient to set `APP_ENV` variable.

File: pkg/pubsub/testdata/config/pubsub.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      - "localhost:9092"
+    # Set by APP_PUBSUB_KAFKA_CLIENT_ID env variable
+    client_id: "test-app"
+    # Set by APP_PUBSUB_KAFKA_CERT_PEM env variable
+    cert_pem: "pem string"
+    # Set by APP_PUBSUB_KAFKA_CERT_PEM_KEY env variable
+    cert_pem_key: "pem key"
+    # Set by APP_PUBSUB_KAFKA_MAX_ATTEMPTS env variable
+    security_protocol: "ssl"
+    enable_ssl_verfification: true
```
Comment (terranisu) [inline pkg/pubsub/testdata/config/pubsub.yml:18]: it's a typo here in the word `verification`

Comment (terranisu) [reply pkg/pubsub/testdata/config/pubsub.yml:18]: Also, can we rename this variable to `ssl_enabled`? In that case, we would have a consistent naming format for the boolean variables - `ssl_enabled`, `publisher_enabled`, `subscriber_enabled`.

Comment (Neurostep) [reply pkg/pubsub/testdata/config/pubsub.yml]: @terranisu fixed both remarks 👍

File: pkg/pubsub/testdata/config/pubsub.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (terranisu) [reply pkg/pubsub/testdata/config/pubsub.yml:28]: @Neurostep What's your take on moving all default settings to the `common` section and populating the development and test environments from there? Similarly what we do in [`go-chassis`](https://github.com/scribd/go-chassis/tree/main/config).

[the remark this answers is not in the document]
Comment (Neurostep) [reply pkg/pubsub/testdata/config/pubsub.yml:28]: @terranisu TBH there is no real value in that as we just test the actual values pass, not a structure of the `yaml` file. Moreover, here we are also testing the `merge` case (from empty to actual values).

[the remark this answers is not in the document]
Comment (terranisu) [reply pkg/pubsub/testdata/config/pubsub.yml:28]: got it.

File: pkg/pubsub/testdata/config/pubsub.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    # Set by APP_PUBSUB_KAFKA_BROKER_URLS env variable
+    broker_urls:
+      - "localhost:9092"
+    # Set by APP_PUBSUB_KAFKA_CLIENT_ID env variable
+    client_id: "test-app"
+    # Set by APP_PUBSUB_KAFKA_CERT_PEM env variable
+    cert_pem: "pem string"
+    # Set by APP_PUBSUB_KAFKA_CERT_PEM_KEY env variable
+    cert_pem_key: "pem key"
+    # Set by APP_PUBSUB_KAFKA_MAX_ATTEMPTS env variable
```
Comment (terranisu) [inline pkg/pubsub/testdata/config/pubsub.yml:16]: It seems this comment should be placed next to line 20. Can we (re-)move it?

File: pkg/pubsub/config.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	config := &Config{}
+	viperBuilder := cbuilder.New("pubsub")
+
+	vConf, err := viperBuilder.Build()
+	if err != nil {
+		return config, err
+	}
+
+	if err = vConf.Unmarshal(config); err != nil {
+		return config, fmt.Errorf("unable to decode into struct: %s", err.Error())
```
Comment (terranisu) [inline pkg/pubsub/config.go:51]: nitpick: Shouldn't the message start with a capital letter?

**Review by terranisu — DISMISSED** [review]
Looks good to me

File: pkg/pubsub/config.go (hunk 5 above)
Comment (Neurostep) [reply pkg/pubsub/config.go:51]: @terranisu It is [recommended](https://github.com/golang/go/wiki/CodeReviewComments#error-strings) to not use capital letters in the error messages

>Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use fmt.Errorf("something bad") not fmt.Errorf("Something bad"), so that log.Printf("Reading %s: %v", filename, err) formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.

**Review by terranisu — APPROVED** [review]
